DSCResources/MSFT_xDatabase/MSFT_xDatabase.psm1

```diff
@@ -252,26 +272,119 @@ function DeployDac([string] $databaseName, [string]$connectionString, [string]$s
         $defaultDacPacApplicationVersion = $defaultDacPacApplicationVersion
     }
 
+    Write-Verbose "DeployDac: Loading DacFx"
+
     Load-DacFx -sqlserverVersion $sqlserverVersion
 
-    $dacServicesObject = new-object Microsoft.SqlServer.Dac.DacServices ($connectionString)
+    Write-Verbose "DeployDac: Loading DacPac"
 
-    $dacpacInstance = [Microsoft.SqlServer.Dac.DacPackage]::Load($dacpacPath)
+    try
+    {        
+        $dacpacInstance = Load-Dacpac $path
+    }
+    catch
+    {
+        Write-Verbose("Unable to load dacpac, error: $_")  
```

Comment:
basically,
If the error is terminating, there is a bug to create guidance (with most of the guidance on what to do here)
https://github.com/PowerShell/DscResources/issues/81 

Write-error is ok if the error is non-terminating (execution of the resource should continue, but execution of dependent resources should stop), with an errorrecord, like created in the guidance.